nixos/modules/services/web-apps/invidious.nix

```diff
@@ -162,14 +203,28 @@ let
       external_port = 80;
     };
 
-    services.nginx = {
+    services.nginx = let
+      ip = if cfg.address == "0.0.0.0" then "127.0.0.1" else cfg.address;
+    in
+    {
       enable = true;
       virtualHosts.${cfg.domain} = {
-        locations."/".proxyPass = "http://127.0.0.1:${toString cfg.port}";
+        locations."/".proxyPass =
+          if cfg.serviceScale == 1 then
+            "http://${ip}:${toString cfg.port}"
```

Comment:
I don't think you can proxy to 0.0.0.0, unless this is something special in nginx but with normal routing in mind this should fail. Did you explicitly test this?